packages/suite/src/hooks/wallet/useSendFormImport.ts

```diff
@@ -41,7 +42,15 @@ export const useSendFormImport = ({
         if (!result || result.length < 1) return; // cancelled
 
         const rates: { currency: string; rate?: number }[] = [];
-        const currencies = result.map(it => it.currency.toLowerCase());
+
+        let currencies;
+        try {
+            currencies = result.map(it => it.currency.toLowerCase());
+        } catch {
+            dispatch(notificationsActions.addToast({ type: 'could-not-parse-csv' }));
+
+            return;
+        }
```

Comment:
`let` must have type declaration, otherwise it implicitly types as `any` But that's besides the point..
The try catch seems very weird. How could .map + .toLowerCase() throw error?
I see that at this point, `result` is typed as `{ [p: string]: string }[]`
So it may contain any properties →  `it.currency` may be undefined, and that's the only way where I would see this code throwing.
So let's look at that explicitly, instead of try catch:
```suggestion

        if (result.some(it => !Object.prototype.hasOwnProperty.call(it, 'currency'))) {
            dispatch(notificationsActions.addToast({ type: 'could-not-parse-csv' }));

            return;
        }
        const currencies = result.map(it => it.currency.toLowerCase());
```

Reply:
I didn't want to iterate twice but I have to agree that being explicit is better. 2c1ca42618cd3b0b5fcf3a01960c5824ab50dd37